src/DssProxyActionsStETH.t.sol

```diff
@@ -0,0 +1,320 @@
+        this.lockStETH(address(wstETHJoin), cdp, 2.756 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), wstETH.getWstETHByStETH(2.756 ether));
+        // Due to wrap precision loss the stored wstETH is equivalent to 1 wei less of stETH
+        assertEq(wstETH.getStETHByWstETH(ink("WSTETH", manager.urns(cdp))), 2.756 ether - 1);
+        this.freeStETH(address(wstETHJoin), cdp, 1.23 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), wstETH.getWstETHByStETH(1.526 ether));
+        // Due to unwrap precision loss the returned stETH is 1 wei less than requested
+        assertEq(stETH.balanceOf(address(this)), prevBalance - 1.526 ether - 1);
+    }
+
+    function testLockStETHAndDraw() public {
+        uint256 cdp = this.open(address(manager), "WSTETH", address(proxy));
+        stETH.approve(address(proxy), 2 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), 0);
+        assertEq(dai.balanceOf(address(this)), 0);
+        uint256 prevBalance = stETH.balanceOf(address(this));
+        this.lockStETHAndDraw(address(jug), address(wstETHJoin), address(daiJoin), cdp, 2 ether, 10 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), wstETH.getWstETHByStETH(2 ether));
+        assertEq(dai.balanceOf(address(this)), 10 ether);
+        assertEq(stETH.balanceOf(address(this)), prevBalance - 2 ether);
+    }
+
+    function testOpenLockStETHAndDraw() public {
+        stETH.approve(address(proxy), 2 ether);
+        assertEq(dai.balanceOf(address(this)), 0);
+        uint256 prevBalance = stETH.balanceOf(address(this));
+        uint256 cdp = this.openLockStETHAndDraw(address(jug), address(wstETHJoin), address(daiJoin), "WSTETH", 2 ether, 10 ether);
+        assertEq(ink("WSTETH", manager.urns(cdp)), wstETH.getWstETHByStETH(2 ether));
+        assertEq(dai.balanceOf(address(this)), 10 ether);
+        assertEq(stETH.balanceOf(address(this)), prevBalance - 2 ether);
+    }
+
+    function testWipeAndFreeStETH() public {
+        uint256 cdp = this.open(address(manager), "WSTETH", address(proxy));
+        stETH.approve(address(proxy), 2 ether);
+        this.lockStETHAndDraw(address(jug), address(wstETHJoin), address(daiJoin), cdp, 2 ether, 10 ether);
+        dai.approve(address(proxy), 8 ether);
+        uint256 prevBalance = stETH.balanceOf(address(this));
+        this.wipeAndFreeStETH(address(wstETHJoin), address(daiJoin), cdp, 1.5 ether, 8 ether);
+        // Due to unwrap precision loss an extra 1 wei of ink resides in the vault
+        assertEq(ink("WSTETH", manager.urns(cdp)), wstETH.getWstETHByStETH(0.5 ether) + 1);
+        assertEq(art("WSTETH", manager.urns(cdp)), 2 ether);
+        assertEq(dai.balanceOf(address(this)), 2 ether);
+        // Due to unwrap precision loss the returned stETH is 1 wei less than requested
+        assertEq(stETH.balanceOf(address(this)), prevBalance + 1.5 ether - 1);
+        assertEq(dai.balanceOf(address(proxy), 2 ether);
```

Comment:
Syntax error here, but also this value is 0 not 2 ether. The DAI remains in the user account not in the proxy. I'd actually remove this line directly.

Reply:
sorry, you're right, i missed line 254 and was coding in the browser :-(